### PR TITLE
Add ECK spec files

### PR DESCRIPTION
**Pull request by ChrsMark**
This PR adds the spec files to provide an e2e setup of ES-Kibana (ECK) along with Metricbeat and Filebeat.

The related issue is #13305

What one can do using these files:
1. Deploy an `es-kibana` cluster using the official ECK operator.
2. Deploy `metricbeat` on k8s
3. Deploy `filebeat` on k8s

Note that Metricbeat is configured to automatically load the pre-built dashboards to Kibana.

cc: @exekias @odacremolbap

### CODE REVIEW

File: deploy/kubernetes/eck/es_kibana.yaml [hunk 1]
```diff
@@ -0,0 +1,40 @@
+# This sample sets up a an Elasticsearch cluster along with a Kibana instance,
+# configured to be able to communicate with each other
+apiVersion: elasticsearch.k8s.elastic.co/v1alpha1
+kind: Elasticsearch
+metadata:
+  name: elasticsearch-sample
+  namespace: kube-system
```
Comment (pebrc) [inline deploy/kubernetes/eck/es_kibana.yaml]: Is there a particular reason why you are deploying everything into `kube-system`? I always thought of it as owned by k8s itself.

File: deploy/kubernetes/eck/es_kibana.yaml [hunk 2]
```diff
@@ -0,0 +1,40 @@
+# This sample sets up a an Elasticsearch cluster along with a Kibana instance,
+# configured to be able to communicate with each other
+apiVersion: elasticsearch.k8s.elastic.co/v1alpha1
```
Comment (pebrc) [inline deploy/kubernetes/eck/es_kibana.yaml]: We have recently released ECK 1.0.0-beta1 which changes the API version and some of the attributes. We tried to document here https://www.elastic.co/guide/en/cloud-on-k8s/1.0/k8s-upgrading-eck.html#k8s-convert-manifests what changed.

File: deploy/kubernetes/eck/es_kibana.yaml (hunk 1 above)
Comment (ChrsMark) [reply deploy/kubernetes/eck/es_kibana.yaml]: Hey @pebrc , sorry for the late response here! The reason for this is that wanted to have everything in the same namespace (ES, metricbeat, kube-state-metrics). So the limitation originated from the `kube-state-metrics`' namespace. In which namespace do you deploy `kube-state-metrics`?

**Comment by ChrsMark** [comment]
Closing this since Sample files were added in https://github.com/elastic/cloud-on-k8s/pull/2041

**Pull request closed without merging**